.github/workflows/link-checker.yml

```diff
@@ -0,0 +1,56 @@
+name: Check documentation links
+
+on:
+  workflow_call:
+    inputs:
+      fail-fast:
+        required: false
+        default: true
+        type: boolean
+  schedule:
+    # runs once a day at 01:00 UTC (adjust as needed)
+    - cron: '0 1 * * *'
+
+permissions:
+  contents: read
+  issues: write
+
+jobs:
+  linkChecker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set fail-fast flag depending on trigger
+        id: flags
+        run: |
+          if [ "${{ github.event_name }}" = "schedule" ]; then
+            echo "fail_fast=false" >> $GITHUB_OUTPUT
+          else
+            echo "fail_fast=${{ inputs.fail-fast }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v2.0.2
+        with:
+          fail: ${{ steps.flags.outputs.fail_fast }}
+          args: |
+            --verbose
+            --exclude-mail
+            --exclude "localhost"
+            --exclude "127.0.0.1"
+            --exclude "^file://"
+            --exclude "github.com/0xMiden/.*"
+            docs/src/**/*.md
+            **/*.md
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
We cannot run this as a `make` command because we want to open an issue based on the result, right?